extensions/data-plane/data-plane-azure-storage/src/main/java/org/eclipse/edc/connector/dataplane/azure/storage/DestinationBlobName.java

```diff
@@ -0,0 +1,41 @@
+/*
+ *  Copyright (c) 2022 Microsoft Corporation
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Marco Primo
+ *
+ */
+
+package org.eclipse.edc.connector.dataplane.azure.storage;
+
+import org.eclipse.edc.util.string.StringUtils;
+
+public class DestinationBlobName {
+    private String folderName;
+    private String blobName;
+
+    public void setBlobName(String blobName) {
+        this.blobName = blobName;
+    }
+
+    public void setFolderName(String folderName) {
+        this.folderName = folderName;
+    }
```

Comment:
please note that in this way this component has a mutable state, and so it's not thread safe (because it's generated at startup and then being used by multiple `DataSink`).

Please make the component immutable and instantiate it once for every DataSink created